packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -176,25 +176,29 @@ export class ServiceBuilderImpl implements ServiceBuilder {
       : createHttpServer(app, logger);
 
     return new Promise((resolve, reject) => {
-      app.on('error', e => {
-        logger.error(`Failed to start up on port ${port}, ${e}`);
+      function handleStartupError(e: unknown) {
+        server.close();
         reject(e);
-      });
+      }
+
+      app.on('error', handleStartupError);
+      server.on('error', handleStartupError);
 
-      const stoppableServer = stoppable(
-        server.listen(port, host, () => {
-          logger.info(`Listening on ${host}:${port}`);
-        }),
-        0,
-      );
+      server.listen(port, host, () => {
+        app.off('error', handleStartupError);
+        server.off('error', handleStartupError);
 
-      useHotCleanup(this.module, () =>
-        stoppableServer.stop((e: any) => {
-          if (e) console.error(e);
-        }),
-      );
+        const stoppableServer = stoppable(server, 0);
 
-      resolve(stoppableServer);
+        useHotCleanup(this.module, () =>
```

Comment:
Afaik it's safer to have this outside the callback in case listen doesn't get called. Works just fine to create the stoppable server outside anyway right?

Reply:
Yeah for sure, moved this outside